sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java

```diff
@@ -957,61 +759,132 @@ void testRequirementC14() throws NoTestData, IOException {
+            final GetStatesOfReportParts getStatesOfReportParts,
+            final MdibHistorian.HistorianResult history,
+            final MdibHistorian.HistorianResult historyNext) {
+        RemoteMdibAccess mdibAccess = history.next();
+        RemoteMdibAccess mdibAccessAhead = historyNext.next();
+
+        final var minimumMdibVersion = ImpliedValueUtil.getMdibVersion(mdibAccess.getMdibVersion());
+        try (final var reports = mdibHistorian.getAllUniqueReports(sequenceId, minimumMdibVersion)) {
+            for (final Iterator<AbstractReport> iterator = reports.iterator(); iterator.hasNext(); ) {
+                final AbstractReport report = iterator.next();
+
+                if (reportClass.isInstance(report)) {
+                    acceptableSequenceSeen.incrementAndGet();
+                    final var reportMdibVersion = ImpliedValueUtil.getReportMdibVersion(report);
+                    // fast-forward history to the mdib version before the report
+                    while (ImpliedValueUtil.getMdibVersion(mdibAccessAhead.getMdibVersion())
+                                    .compareTo(reportMdibVersion)
+                            < 0) {
+                        mdibAccessAhead = historyNext.next();
+                        mdibAccess = history.next();
+                    }
 
-                            for (var reportPart : operationalStateReport.getReportPart()) {
-                                for (var operationState : reportPart.getOperationState()) {
-                                    final var stateBeforeReport = first.getState(
-                                            operationState.getDescriptorHandle(), AbstractOperationState.class);
-                                    final var stateAfterReport = second.getState(
-                                            operationState.getDescriptorHandle(), AbstractOperationState.class);
-                                    assertTrue(
-                                            stateBeforeReport.isPresent() && stateAfterReport.isPresent(),
-                                            String.format(STATE_ABSENT, operationState.getDescriptorHandle()));
-                                    assertNotEquals(
-                                            stateAfterReport.orElseThrow(),
-                                            stateBeforeReport.orElseThrow(),
-                                            String.format(STATE_UNCHANGED, operationState.getDescriptorHandle()));
-                                }
+                    final var reportParts = getStatesOfReportParts.apply(reportClass.cast(report));
+                    for (var reportPart : reportParts) {
+                        for (var state : reportPart) {
+                            final Optional<? extends AbstractState> stateBeforeReport;
+
+                            stateBeforeReport = mdibAccess.getState(state.getDescriptorHandle(), stateClass);
```

Comment:
The previous version that made the distinction between `AbstractMultiState` and `AbstractState` was better, how can you retrieve the correct multi state handle now? This only queries by descriptor handle.

Reply:
True, changed it back